src/votekit/ballot_graph.py

```diff
@@ -0,0 +1,226 @@
+
+
+Graphs = {}
+
+
+class BallotGraph():
+    profile: PreferenceProfile
+    ballot_dict: dict
+    num_cands: int
+    num_voters: int
+    
+    
+    def __init__(self,profile):
+        self.profile = profile
+        self.ballot_dict= profile.to_dict()
+        
+        #cand_list = []
+        #for key in ballot_dict.keys():
+        #    for i in key:
+        #        cand_list.append(i)
+      
+        self.num_cands = len(self.profile.get_candidates())
+        
+        if self.num_cands not in Graphs.keys():
+            build_graph(self.num_cands)
+        
+        
+        all_ballots = Graphs[self.num_cands].nodes
+        di = {}
+        for ballot in all_ballots:
+            di[ballot] = 0
+        
+        self.ballot_dict = di | ballot_dict
+        
+        
+        self.clean()
+        
+        self.num_voters = sum(ballot_dict.values())
+        
+        
+    def clean(self): #deletes empty ballots, changes n-1 length ballots to n length ballots and updates counts
+        di = self.ballot_dict.copy()
+        
+        for ballot in di.keys():
+            if len(ballot)==0:
+                self.ballot_dict.pop(ballot)
+            elif len(ballot)==self.num_cands-1:
+                for i in self.profile.get_candidates():
+                    if i not in ballot:
+                        self.ballot_dict[ballot+tuple([i])]+= di[ballot]    
+                        self.ballot_dict.pop(ballot)
+                        break
+                
+        
+    def visualize(self, neighborhoods = {}):
+        Gc = Graphs[self.num_cands]    
+        if neighborhoods == {}:
+            self.clean()
+            WHITE = (1,1,1)
+            BLACK=(0,0,0)
+            cols = get_colors(self.num_cands, [WHITE,BLACK])
+            node_cols = []
+            ballots = list(Gc.nodes)
+        
+            for bal in Gc.nodes:
+                if self.ballot_dict[bal]!=0:
+                    i = self.profile.get_candidates().index(bal[0])
+                    node_cols.append(cols[i])
+                else:
+                    node_cols.append(WHITE)
+            ##want to include number of votes as part of labels,  color ballots with 0 votes grey
+        
+            nx.draw_networkx(Gc,with_labels = True, node_color = node_cols)
+            
+        else:
+            WHITE = (1,1,1)
+            BLACK = (0,0,0)
+            cols = get_colors(len(neighborhoods), [WHITE, BLACK])
+            node_cols = []
+            centers = list(neighborhoods.keys())
+            
+            for bal in Gc.nodes:
+                found = False
+                for i in range(len(centers)):    
+                    weight = (neighborhoods[centers[i]])[1]
+                    if bal in (neighborhoods[centers[i]])[0].nodes:
+                        node_cols.append(tuple([(1-self.ballot_dict[bal]/weight)*x for x in cols[i]]))
+                        found = True
+                        break ##breaks the inner for loop
```

Comment:
comments should focus on the "why" not the "what", a person reading this would presumably know that break exits a for loop, explaining why that's happening is more useful/important than saying what break does here